### PR TITLE
Medical - Add dropping weapons on unconsciousness

### PR DESCRIPTION
**Pull request by AndreasBrostrom**
**When merged this pull request will:**
- Added setting to allow a change of dropping held weapon when going unconscious (Default 0% chance / off)
- Moved `throwWeapon` function from `hitreactions` to `common`.

### CODE REVIEW

**Comment by BrettMayson** [comment]
Do we also want a slider to set the chance of this happening?

File: addons/medical/functions/fnc_setUnconscious.sqf [hunk 1]
```diff
@@ -46,6 +46,14 @@ if (currentWeapon _unit != primaryWeapon _unit) then {
     _unit selectWeapon primaryWeapon _unit;
 };
 
+if (QGVAR(dropWeaponOnUnconscious)) then {
+    if (_unit == ACE_player) then {
+        if (_knockOut) then {
+            player call EFUNC(hitreactions,throwWeapon);
```
Comment (commy2) [inline addons/medical/functions/fnc_setUnconscious.sqf:52]: Why should the Zeus throw their weapon when the remote controlled AI gets shot?

Comment (ColdEvul) [reply addons/medical/functions/fnc_setUnconscious.sqf:52]: Hehe, they should not.

Assuming that this should be suffisant.
```suggestion
            ACE_player call EFUNC(hitreactions,throwWeapon);
```
 also dont want ai to drop their weapons cause ai

**Comment by ColdEvul** [comment]
> Do we also want a slider to set the chance of this happening?

Sure i can hammer one. The setting is off per default.

**Comment by kymckay** [comment]
>Do we also want a slider to set the chance of this happening?

I'd take out the "also", boolean + slider should almost always just be a single slider setting that captures the "true" and "false" states anyway

**Comment by kymckay** [comment]
I'd also point out that this PR introduces a dependency on hitreactions which is not reflected in the config

I'd also point out this is only the public function, it is not called from the statemachine and as such this behaviour won't be seen unless calling the public function, the setting doesn't really belong in this component which houses public APIs. It will need to effect the underlying unconscious state code.

**Comment by ColdEvul** [comment]
> I'd also point out this is only the public function, it is not called from the statemachine and as such this behaviour won't be seen unless calling the public function, the setting doesn't really belong in this component which houses public APIs. It will need to effect the underlying unconscious state code.

were is this defined?

**Comment by Salluci** [comment]
Logic for this needs to be moved/added to [addons/medical_status/functions/fnc_setUnconsciousState.sqf](https://github.com/acemod/ACE3/blob/master/addons/medical_status/functions/fnc_setUnconsciousState.sqf) so it's used internally and actually has an effect on default gameplay.

**Comment by ColdEvul** [comment]
Will 

> Logic for this needs to be moved/added to [addons/medical_status/functions/fnc_setUnconsciousState.sqf](https://github.com/acemod/ACE3/blob/master/addons/medical_status/functions/fnc_setUnconsciousState.sqf) so it's used internally and actually has an effect on default gameplay.

Will adjust

File: addons/medical/functions/fnc_setUnconscious.sqf [hunk 2]
```diff
@@ -46,6 +46,16 @@ if (currentWeapon _unit != primaryWeapon _unit) then {
...
 
+if (GVAR(dropWeaponOnUnconscious) != 0) then {
+    if (_unit == ACE_player) then {
+        if (_knockOut) then {
+            if (random 1 <= GVAR(dropWeaponOnUnconscious)) then {
+                ACE_player call EFUNC(hitreactions,throwWeapon);
+            };
+        };
+    };
+};
```
Comment (severgun) [inline addons/medical/functions/fnc_setUnconscious.sqf:57]: Why not to use &&? And there is already `if (_knockOut)` couple lines lower. So why not to put it inside?
```sqf
if (GVAR(dropWeaponOnUnconscious) != 0 && {_unit == ACE_player && {random 1 <= GVAR(dropWeaponOnUnconscious)}}) then {
    ACE_player call EFUNC(hitreactions,throwWeapon);
};
```

File: addons/medical/stringtable.xml [hunk 3]
```diff
@@ -18,6 +18,12 @@
             <Turkish>ACE Medikal</Turkish>
             <Japanese>ACE 医療</Japanese>
         </Key>
+        <Key ID="STR_ACE_Medical_EnableDropWeaponOnUnconscious_DisplayName">
+            <English>Drop weapon on unconscious</English>
```
Comment (severgun) [inline addons/medical/stringtable.xml:22]: Should be Title Case and mention that this is **chance**

File: addons/medical/functions/fnc_setUnconscious.sqf (hunk 2 above)
Comment (AndreasBrostrom) [reply addons/medical/functions/fnc_setUnconscious.sqf:57]: agree way better will adjust.

File: addons/medical/stringtable.xml (hunk 3 above)
Comment (commy2) [reply addons/medical/stringtable.xml:22]: Drop Weapon When Falling Unconscious

Drop Weapon When Losing Consciousness

🤔

Comment (Drofseh) [reply addons/medical/stringtable.xml:22]: Drop Weapon On Loss of Consciousness

**Comment by AndreasBrostrom** [comment]
ToDo
- [ ] Change name of setting display name
- [ ] Rewrite and move to: addons/medical_status/functions/fnc_setUnconsciousState.sqf
- [ ] Test

**Comment by Salluci** [comment]
I don't like the idea of making `medical_status` dependent on `hitreactions`.

**Comment by jonpas** [comment]
Please don't force push, it is impossible to follow your changes.

**Comment by AndreasBrostrom** [comment]
> I don't like the idea of making `medical_status` dependent on `hitreactions`. Rather move `fnc_throwWeapon` to `common` component. Would be more useful there for this and other modules (throwing weapons when surrendering, for example).
> 
> Variable name/stringtable spelling is wrong. "Chance", not "Chanse".

I'll make the adjustment

> Please don't force push, it is impossible to follow your changes.

Did due to rebase, retore and squashing wip commits. It was eather that or a new PR.

File: addons/medical_status/stringtable.xml [hunk 4]
```diff
@@ -111,5 +111,11 @@
             <Spanish>Controla la rapidez con que fluye el líquido de las bolsas intravenosas. El cambio de volumen de la bolsa IV se calcula como: \n intervalo (s) de tiempo * cambio iv por segundo (4.1667 mL/s) * velocidad de flujo (este coeficiente).</Spanish>
             <Turkish>IV Torbalardan sıvının ne kadar hızlı aktığını kontrol eder. IV Torba hacim değişikliği şu şekilde hesaplanır: zaman aralıkları iv saniye başına değişim (4.1667 mL / s) akış hızı (bu katsayı).</Turkish>
         </Key>
+        <Key ID="STR_ACE_Medical_Status_DropWeaponUnconsciousChanse_DisplayName">
+            <English>Chance to Drop Weapon On Loss of Consciousness</English>
+        </Key>
+        <Key ID="STR_ACE_Medical_Status_DropWeaponUnconsciousChanse_Description">
+            <English>The probability of a person of dropping their weapon on loss of concussion.</English>
```
Comment (rautamiekka) [inline addons/medical_status/stringtable.xml:118]: A thinking error there: `concussion` -> `consciousness`.

**Review by rautamiekka — CHANGES_REQUESTED** [review]

File: addons/medical_status/stringtable.xml (hunk 4 above)
Comment (AndreasBrostrom) [reply addons/medical_status/stringtable.xml:118]: was not conscious

**Comment by AndreasBrostrom** [comment]
Updated header

**Review by LinkIsGrim — CHANGES_REQUESTED** [review]
Need to update function call in https://github.com/acemod/ACE3/blob/49f4ea23e70dd4784b28248f605b14105f4a9a24/addons/fire/functions/fnc_burnReaction.sqf#L24

File: addons/medical_status/functions/fnc_setUnconsciousState.sqf [hunk 5]
```diff
@@ -42,6 +42,11 @@ if (_active) then {
         _unit setVariable [QEGVAR(medical,lastWakeUpCheck), _lastWakeUpCheck max CBA_missionTime];
     };
 
+    // Player drop weapon chance
```
Comment (LinkIsGrim) [inline addons/medical_status/functions/fnc_setUnconsciousState.sqf:45]:
```suggestion
    // Chance for player to drop weapon
```

File: addons/medical_status/stringtable.xml [hunk 6]
```diff
@@ -111,5 +111,11 @@
             <Spanish>Controla la rapidez con que fluye el líquido de las bolsas intravenosas. El cambio de volumen de la bolsa IV se calcula como: \n intervalo (s) de tiempo * cambio iv por segundo (4.1667 mL/s) * velocidad de flujo (este coeficiente).</Spanish>
             <Turkish>IV Torbalardan sıvının ne kadar hızlı aktığını kontrol eder. IV Torba hacim değişikliği şu şekilde hesaplanır: zaman aralıkları iv saniye başına değişim (4.1667 mL / s) akış hızı (bu katsayı).</Turkish>
         </Key>
+        <Key ID="STR_ACE_Medical_Status_DropWeaponUnconsciousChance_DisplayName">
+            <English>Unconscious Drop Weapon</English>
```
Comment (LinkIsGrim) [inline addons/medical_status/stringtable.xml:115]:
```suggestion
            <English>Unconsciousness Weapon Drop Chance</English>
```

File: addons/medical_status/stringtable.xml [hunk 7]
```diff
@@ -111,5 +111,11 @@
...
+        </Key>
+        <Key ID="STR_ACE_Medical_Status_DropWeaponUnconsciousChance_Description">
+            <English>The probability of a player dropping their weapon on loss of consciousness.</English>
```
Comment (LinkIsGrim) [inline addons/medical_status/stringtable.xml:118]:
```suggestion
            <English>Chance for a player to drop their weapon when going unconscious.\nHas no effect on AI.</English>
```

**Review by LinkIsGrim — COMMENTED** [review]
Either that or a better setting title. Otherwise it's good.

**Review by LinkIsGrim — APPROVED** [review]

File: addons/fire/functions/fnc_burnReaction.sqf [hunk 8]
```diff
@@ -21,7 +21,7 @@ if (
     && {_unit in _unit && { !(currentWeapon _unit isEqualTo "") }}
     && {!isPlayer _unit || GVAR(dropWeapon >= 2)}
```
Comment (PabstMirror) [inline addons/fire/functions/fnc_burnReaction.sqf:22]: it was fine but makes me very uncomfortable
```suggestion
    && {!isPlayer _unit || GVAR(dropWeapon) >= 2}
```

Comment (LinkIsGrim) [reply addons/fire/functions/fnc_burnReaction.sqf:22]: that... shouldn't work?

**Pull request merged**